### PR TITLE
Fix: Repeated .fit() calls ignore max_steps iteration bound 

### LINKED ISSUE #4809: Repeated .fit() calls ignore max_steps iteration bound
## 🐛 Bug

Hello!
While trying to convert my code to PL (I'm starting to become a big fan!) I came across some unexpected behavior: In an iteration-based training setup repeated calls of `trainer.fit()` result in ignoring the iteration bound set by the `max_steps` argument. The trainer will finish the entire epoch, even though in my opinion it shouldn't (forgive me if I missed something obvious, which is easily possible since I'm new to PL).

## Please reproduce using [the BoringModel and post here](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing)

https://colab.research.google.com/drive/1gKLNoYXjW7s3ifSJJ00SVZi4b08GDy5F?usp=sharing

### To Reproduce

In the BoringModel, I only changed the test cell to something like this:
```
def test_x(tmpdir):
    # init model
    model = BoringModel()

    iterations = 80
    print('fit() should only do {} iterations'.format(iterations))

    # Initialize a trainer
    trainer = pl.Trainer(
        max_steps=iterations, 
        progress_bar_refresh_rate=20
    )

    # Train the model ⚡
    trainer.fit(model, train, val)  # here the trainer correctly does 80 iterations

    print('global_step: ', trainer.global_step)
    print('first fit() done')

    trainer.fit(model, train, val)  # in this fit(), the trainer will complete the epoch

    # trainer.test(test_dataloaders=test)
```
### Expected behavior

I expect repeated `trainer.fit()` calls to result in either
1. No action, because `trainer.global_step == trainer.max_steps`

or

2. Again fitting of the model for another `max_steps` iterations

I think most people would expect the former.

### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 1.0.7
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

### Additional context

The same problem arises if a fully trained model is loaded, and then `trainer.fit()` is called. This is especially troubling when a `trainer.test()` follows.

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes #4809 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: tests/callbacks/test_progress_bar.py [hunk 1]
```diff
@@ -146,8 +146,6 @@ def test_progress_bar_fast_dev_run(tmpdir):
     assert 1 == progress_bar.total_train_batches
     # total val batches are known only after val dataloaders have reloaded
 
-    trainer.fit(model)
-
```
Comment (rohitgr7) [inline tests/callbacks/test_progress_bar.py:150]: why remove this?

**Review by rohitgr7 — APPROVED** [review]
looks good!

File: tests/callbacks/test_progress_bar.py (hunk 1 above)
Comment (kaushikb11) [reply tests/callbacks/test_progress_bar.py:150]: When the User would use the `fast_dev_run=True` flag, they won’t be able to call `.fit()` twice or more, as it would reach the `max_steps`.

Comment (rohitgr7) [reply tests/callbacks/test_progress_bar.py:150]: oh okay, it's on fast_dev_run.. didn't see that.

**Review by Borda — APPROVED** [review]
